Supporting cordova-cli@9 or cordova-lib@9

sagrawal31 opened the pull request:
As per `cordova-lib@9`, the `requireCordovaModule` can only be used to get the `cordova-*` dependency. This PR is to support that change.

For more info, https://github.com/apache/cordova-lib/issues/706

dpa99c committed:
Remove redundant build-extras.gradle

dpa99c committed:
Remove unnecessary extra <config-file> block which can lead to race condition

sagrawal31 committed:
Supporting cordova-cli@9 or cordova-lib@9

ilyakamens commented:
Would love for this to be merged.

forgr-owner commented:
Would love it too !

mismith commented:
@sagrawal31 can you get the build passing?

sagrawal31 commented:
I’ll try.

On Wed, 3 Apr 2019 at 8:23 AM, Murray Smith <notifications@github.com>
wrote:

> @sagrawal31 <https://github.com/sagrawal31> can you get the build passing?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/arnesson/cordova-plugin-firebase/pull/1034#issuecomment-479311342>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABuI4jp56Ptjqs8HzN5EH1osIKxk5ENtks5vdBelgaJpZM4cNGkF>
> .
>
-- 
Thanks,
Shashank Agrawal
Co-Founder & CTO
Wiz Panda Techno Solutions
+91 94252-20659
shashank@wizpanda.com

-- 
DISCLAIMER: The information in this message is confidential and may be 
legally privileged. It is intended solely for the addressee. Access to this 
message by anyone else is unauthorized. If you are not the intended 
recipient, any disclosure, copying, or distribution of the message, or any 
action or omission taken by you in reliance on it, is prohibited and may be 
unlawful. Please immediately contact the sender if you have received this 
message in error. Further, this e-mail may contain viruses and all 
reasonable precaution to minimize the risk arising there from is taken by 
Wiz Panda. Wiz Panda is not liable for any damage sustained by you as a 
result of any virus in this e-mail. All applicable virus checks should be 
carried out by you before opening this e-mail or any attachment thereto.

Thank you - Wiz Panda Techno Solutions Private Limited.

sagrawal31 committed:
Trying to fix the build

sagrawal31 committed:
Removing old cordova-android because of a failing test case
https://github.com/arnesson/cordova-plugin-firebase/issues/755

sagrawal31 committed:
Only installing xcode for ios platform

sagrawal31 committed:
Fixing old failing Android tests by not installing the cordova & platform again

sagrawal31 commented:
@mismith I fixed the Travis build. I think, all of them was from before this MR.

mismith commented:
Nice work! @arnesson could this be merged now?

BobCashStory commented:
i tryed your with `cordova plugin add https://github.com/wizpanda/cordova-plugin-firebase` pr and it work well !

felipeclopes commented on test/test-with-3-plugins.sh:
```diff
@@ -9,7 +9,8 @@ ADDITIONAL_PLUGIN_1=$4
 ADDITIONAL_PLUGIN_2=$5
 ADDITIONAL_PLUGIN_3=$6
 
-bash ./test/platform-add.sh $CORDOVA_VERSION $PLATFORM $PLATFORM_VERSION
```
Is this what made the build pass? If yes, I'll submit a separate PR with this fix, so all the broken PRs would start passing after rebasing.

sagrawal31 replied:
Not really, all but yes, it did fixed a few test cases. Since the cordova & platform version passed to this script from the parent script are same hence there was no point reinstalling it again so I commented out that. But I would love to see the detailed error when this script was trying to install the same cordova & platform.

sagrawal31 replied:
I would also love to help you on rebasing PRs & getting them merged which are good to go because I saw various good PRs

forgr-owner replied:
how can we go forward ?

sagrawal31 replied:
@felipeclopes can tell anything

felipeclopes replied:
I'm not really sure hot to proceed. It seems that the first step would be to try to fix the test cases that are broken. If you look at the list of PRs to merge, the only ones passing are the ones who removed the tests for older Cordova versions and this one. 

Maybe a separate PR just with this fix of the build cold help the other PRs to rebase and get a green status on the build system.

sagrawal31 replied:
In my thinking, we can start with merging this branch to master then start rebasing other PR with master so most of the PRs will by default pass

sagrawal31 replied:
@felipeclopes any update? I'll still suggest to move forward with this

pcova2099 commented:
Hi, iIve installed the plugin today (2019-04-15) but the error still happening (Failed to install 'cordova-plugin-firebase': CordovaError: Using "requireCordovaModule" to load non-cordova module "xcode" is not supported. Instead, add this module to your dependencies and use regular "require" to load it.) 

```
    <plugin name="cordova-plugin-firebase" spec="https://github.com/arnesson/cordova-plugin-firebase" />
    <engine name="android" spec="7.1.4" />
    <engine name="ios" spec="4.5.5" />
```

Version 2.0.5 was installed by cordova

forgr-owner commented:
@pcova2099 it's not merged that why, if you want to make it work before merge use this `cordova plugin add https://github.com/wizpanda/cordova-plugin-firebase`

pcova2099 commented:
I got this error with 

> cordova plugin add https://github.com/wizpanda/cordova-plugin-firebase


```
Installing "cordova-plugin-firebase" for ios
Failed to install 'cordova-plugin-firebase': Error: Cannot find module 'xcode'
```
```
Ionic:

   ionic (Ionic CLI)  : 4.4.0 (C:\Users\pcova\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework    : ionic-angular 3.9.3
   @ionic/app-scripts : 3.2.1

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 3.1.2, (and 5 other plugins)

System:

 (C:\Program Files (x86)\Android\android-sdk)
   NodeJS            : v8.11.1 (C:\Program Files (x86)\nodejs\node.exe)
   npm               : 5.6.0
   OS                : Windows 10
```

sagrawal31 commented:
@pcova2099 you need to run `npm i xcode --save-dev` in your project as a temporary workaround as I have mentioned here https://github.com/wizpanda/cordova-plugin-firebase#cordova-lib9-support

gdespiritorflex commented:
+1

dpa99c committed:
Fixes issues cause by Firebase SDK updates on 5 April 2019 (https://firebase.google.com/support/release-notes/android#update_-_april_05_2019) which removed deprecated API features causing Android build failures.
See https://github.com/arnesson/cordova-plugin-firebase/issues/1057

(cherry picked from commit 143f450894e89ff7d77d31197738386d5bea2a16)

hiepxanh commented:
when this will be merge?

remisture commented:
Is this project dead?

dpa99c committed:
Remove obsolete <service> entry for FirebasePluginInstanceIDService.
See https://github.com/arnesson/cordova-plugin-firebase/pull/1058#pullrequestreview-236291796

(cherry picked from commit 25c7a50c2ed9fa28babc7e7090f51f64b5df9304)

paulsmith47 commented:
I am also waiting for this PR to be merged will this happen any time soon?

sagrawal31 committed:
Merge pull request #2 from wizpanda/firebase-changes

Fixes for latest firebase changes

sagrawal31 committed:
3.0.0

sagrawal31 committed:
Updating README.md

sagrawal31 commented:
[We](https://www.wizpanda.com) are now maintaining a fork [wizpanda/cordova-plugin-firebase-lib](https://github.com/wizpanda/cordova-plugin-firebase-lib#difference-from-the-fork-repository) with all these fixes & changes.

pcova2099 commented:
thanks a lots, wizpanda/cordova-plugin-firebase-lib it's working perfectly well.

sagrawal31 committed:
Adding npm version badge.

gdespiritorflex commented:
+1. Please merge this.

sgioia9 commented:
Let's merge this...

Commit:
Added ability to log js stack trace on android.

Commit:
Missed adding file to plugin.xml

Commit:
Fixed error if stacktrace is missing data and js bug if stackTrace was undefined.

Commit:
Fixed ios error logging and added ability to log stack traces.

maximilianloy committed:
Add createChannel function for Android 0

sagrawal31 committed:
Formatted the code

sagrawal31 committed:
Merge pull request #9 from maximilianloy/android-createchannel

Add createChannel function for Android 0

sagrawal31 committed:
Only changing the indentation to 4 spaces.

sagrawal31 committed:
3.1.0

sagrawal31 committed:
Trying to fix the Travis CI build.

sagrawal31 committed:
First version of changes to make the app build after breaking changes rolled out by Google on Jun 17, 2019

sagrawal31 committed:
Changing the minimum required cordova-android version to 8.0.0

distinctdan committed:
Merge branch 'master' into master

sagrawal31 committed:
Merge pull request #8 from distinctdan/master

Fixed crash logging on iOS, added ability to log JS stack traces.

sagrawal31 committed:
Some code cleanup and changes to log stacktrace in Firebase.

sagrawal31 committed:
3.2.0

sagrawal31 committed:
Backward compatibility fixes to support breaking changes released by Firebase on Jun 17, 2019.

sagrawal31 committed:
Merge pull request #14 from wizpanda/firebase-breaking-backward

Backward compatibility fixes to support breaking changes released by Google Firebase on Jun 17, 2019

sagrawal31 committed:
3.3.0

sagrawal31 committed:
Updating README.md

sagrawal31 committed:
Merge pull request #13 from wizpanda/google-breaking-changes

Fix breaking changes released by Google Firebase on Jun 17, 2019

sagrawal31 committed:
GitHub issue template cleanup. Updating JS files to use "const" or "let" instead of "var"

sagrawal31 committed:
4.0.0

sagrawal31 committed:
Formatting plugin.xml to use 4 spaces.

sagrawal31 committed:
Fixed typo in firebase-browser.js fixes #16

sagrawal31 committed:
Only formatting the JS files.

sagrawal31 committed:
4.0.1

sagrawal31 committed:
Breaking Change: Moving to CocoaPods to manage Firebase, Crashlytics & Fabric dependency management for iOS.
This now requires cordova-lib to be minimum 9 and cordova-ios to be minimum 5.0.1.

(A few changes taken from https://github.com/dpa99c/cordova-plugin-firebase/commit/9ae6433371cb2325eb876d68d95a9435efe27cb4#diff-91f698c29d714cf5947fe5e7ae01dbaf)

sagrawal31 committed:
Using optimistic operator to specify Firebase dependencies for iOS.

sagrawal31 committed:
Breaking Change: Merging `logJSError` to `logError` method for consistency and standard.

https://github.com/wizpanda/cordova-plugin-firebase-lib/pull/8#issuecomment-503694374

sagrawal31 committed:
Updating the API doc.

https://github.com/wizpanda/cordova-plugin-firebase-lib/pull/8#issuecomment-503694374

sagrawal31 committed:
Generated "Table of Contents" via markdown-toc generator.

sagrawal31 committed:
Handling a few edge cases for "logError" method.

sagrawal31 committed:
Updating README.

sagrawal31 committed:
Merge pull request #21 from wizpanda/android-fixes

Merging two logError methods, documentation update and some edge case handling

distinctdan committed:
Merge pull request #1 from wizpanda/master

Updating to match master

distinctdan committed:
Added more docs for passing a stacktrace to logError

sagrawal31 committed:
Merge pull request #22 from distinctdan/master

Added more docs for passing a stacktrace to logError

sagrawal31 committed:
4.1.0

sagrawal31 committed:
Updating CHANGELOG & docs.

sagrawal31 committed:
Updating README.

sagrawal31 committed:
More clear documentation about different cordova platform versions.

sagrawal31 committed:
Merge pull request #23 from wizpanda/ios-pods

Using CocoaPods for maintaining Firebase dependencies for iOS

sagrawal31 committed:
5.0.0

sagrawal31 committed:
Updating README.

teknepal committed:
Fix for ampersand in the app name

sagrawal31 committed:
Reducing the Firebase iOS dependencies to 5.x to avoid breaking changes.

sagrawal31 committed:
Removed deprecated method "getInstanceId"

sagrawal31 committed:
Merge pull request #28 from wizpanda/ios-fixes

Downgrading Fireabse iOS dependency to 5.x to keep running existing code

sagrawal31 committed:
5.1.0

teknepal committed:
changed const to var

sagrawal31 committed:
Merge pull request #26 from teknepal/fix_ampersand_app_name

Fix for ampersand in the app name

sagrawal31 committed:
Improving the JS code. Using chalk to highlights a few logs.

sagrawal31 committed:
Some docs improvement.

sagrawal31 committed:
Merge pull request #30 from wizpanda/js-improvement

JavaScript code improvements & documentation

sagrawal31 committed:
Some more logging improvement.

sagrawal31 committed:
5.1.1

sagrawal31 committed:
Using tabular format for different version of plugin.

sagrawal31 committed:
Moving the difference doc on the top & some cleanup.

sagrawal31 committed:
Updating README for diverting developers to new repository.

nascarjake commented:
This is still marked open and I'm still getting the error, assuming this isn't merged yet. Do we know when it will be? Have yet to try the work around. Wanted to know how close the merge was.

ryanmtaylor commented:
@nascarjake a lot of people have moved to https://github.com/wizpanda/cordova-plugin-firebase-lib or https://github.com/dpa99c/cordova-plugin-firebasex
